R/reactable.R

```diff
@@ -809,6 +807,86 @@ renderReactable <- function(expr, env = parent.frame(), quoted = FALSE) {
   htmlwidgets::shinyRenderWidget(expr, reactableOutput, env, quoted = TRUE)
 }
 
+#' Convert a reactable widget to HTML tags
+#'
+#' This S3 method exists to enable [reactable()]'s `static` rendering option.
+#'
+#' @param x a [reactable()] instance.
+#' @param standalone
+#'
+#' @export
+as.tags.reactable <- function(x, standalone = FALSE) {
+  # Should call htmlwidgets:::as.tags.htmlwidget(), which calls htmlwidgets::toHTML()
+  result <- NextMethod("as.tags", x, standalone = standalone)
+
+  # Make sure react is loaded before the reactable dependencies
+  dependencies <- c(
+    list(
+      reactR::html_dependency_corejs(), # Necessary for RStudio Viewer version < 1.2 and IE11
+      reactR::html_dependency_react(),
+      reactR::html_dependency_reacttools()
+    ),
+    htmlDependencies(result)
+  )
+
+  result <- attachDependencies(result, dependencies)
+
+  attribs <- x$x$tag$attribs
+  if (!isTRUE(attribs$static)) {
+    return(result)
+  }
+
+  # TODO: warn if this is running inside renderReactable()?
+  input_json <- toJSON(list(
+    props = attribs,
+    evals = htmlwidgets::JSEvals(attribs)
+  ))
+
+  output <- list()
+
+  tryCatch({
+    ctx <- V8::v8()
+    ctx$source(system.file("htmlwidgets/reactable.server.js", package = "reactable", mustWork = TRUE))
+    output <- ctx$call("Reactable.renderToHTML", input_json)
+  }, error = function(e) {
+    stop("Failed to render table to static HTML:\n", conditionMessage(e), call. = FALSE)
```

Comment:
Could we keep the warning and fallback path when this errors out or V8 isn't installed? Static rendering is still very experimental and not essential functionality either, so I'd rather have this gracefully fall back to the usual client rendering when something goes wrong.